src/ucp/wireup/select.c

```diff
@@ -652,24 +652,33 @@ static double ucp_wireup_fp8_pack_unpack_latency(double latency)
     return UCS_FP8_UNPACK(LATENCY, packed_lat) / UCS_NSEC_PER_SEC;
 }
 
+static double ucp_wireup_fp8_pack_unpack_bw(double bandwidth)
+{
+    ucs_fp8_t packed_bw = UCS_FP8_PACK(BANDWIDTH, bandwidth);
+    return UCS_FP8_UNPACK(BANDWIDTH, packed_bw);
+}
+
 static inline double
-ucp_wireup_tl_iface_latency(ucp_context_h context,
-                            const uct_iface_attr_t *iface_attr,
+ucp_wireup_tl_iface_latency(const ucp_worker_iface_t *wiface,
                             const ucp_address_iface_attr_t *remote_iface_attr)
 {
-    double local_lat, lat_nsec;
+    ucp_context_h context = wiface->worker->context;
+    double local_lat, lat_nsec, lat_lossy;
 
     if (remote_iface_attr->addr_version == UCP_OBJECT_VERSION_V1) {
+        local_lat = ucp_wireup_iface_lat_distance_v1(wiface);
         /* Address v1 contains just latency overhead */
-        return ((iface_attr->latency.c + remote_iface_attr->lat_ovh) / 2) +
-                (iface_attr->latency.m * context->config.est_num_eps);
+        return ((local_lat + remote_iface_attr->lat_ovh) / 2) +
+               (wiface->attr.latency.m * context->config.est_num_eps);
     } else {
+        local_lat = ucp_wireup_iface_lat_distance_v2(wiface);
         /* FP8 is a lossy compression method, so in order to create a symmetric
          * calculation we pack/unpack the local latency as well */
-        lat_nsec  = ucp_tl_iface_latency(context, &iface_attr->latency) *
+        lat_nsec  = local_lat * UCS_NSEC_PER_SEC;
+        lat_lossy = ucp_wireup_fp8_pack_unpack_latency(lat_nsec) /
```

Comment:
does it make sense to move the multiplication and division by UCS_NSEC_PER_SEC inside ucp_wireup_fp8_pack_unpack_latency function?